iXBRLViewerPlugin/viewer/src/js/fact.js

```diff
@@ -267,3 +267,22 @@ Fact.prototype.widerConcepts = function () {
     }
     return concepts;
 }
+
+Fact.prototype.narrowerConcepts = function () {
+    if (!this._report.usesAnchoring()) {
+        return [];
+    }
+    var concepts = [];
+    const wn = this._report.data.rels["w-n"];
+    for (const elr in wn) {
+        for (const src in wn[elr]) {
+            if (src == this.conceptName()) {
+                for (const target of wn[elr][src]) {
```

Comment:
Lookup the key directly rather than iterating?
```
for (const elr in wn) {
  const targets = wn[elr][this.conceptName()];
  if (targets) {
    for (const target of targets) {
```

Reply:
Yeah, not sure what I was thinking there!   Refactored in 3a8172453f244c01872d5d7cdb249599de8f248c (see previous comment).